### PR TITLE
Pass on the dark mode section

### PR DESCRIPTION
**Pull request by pavpanchekha**
This PR passes over the "dark mode" and "color scheme" sections. The major changes are:

- I moved the color scheme section to be right after the dark mode section so we don't have to have two copies of the browser color scheme. I thought that was likely to confuse some readers who know about `@media` rules.
- Clarified that the dark mode section has two parts, the chrome and the page contents.
- I modified `DocumentLayout.paint()` to not draw the background rectangle, instead just leaving that to `draw`. Maybe this is bad because it causes a one-frame wrong draw? I don't notice that, but if this was the underlying reason let me know and we'll do this some other way. (Can the one-frame wrong-draw still occur if the web page is shorter than the browser height?)
- I made another small change to how media queries are parsed, thinking about how the reader might want to extend their browser with more media queries.

### CODE REVIEW

File: book/accessibility.md [hunk 1]
```diff
@@ -698,6 +638,133 @@ sheets, and instead rely on extensions.
 
 :::
 
+Customizing dark mode
```
Comment (chrishtr) [inline book/accessibility.md:641]: Question: I organized the chapter previously around this order:

Section 1. Some a11y challenges, and how the browser addreseses them
Section 2. APIs for developers to customize.

I think this changes that right? Do you think it'd be better to restructure?

File: book/accessibility.md [hunk 2]
```diff
@@ -455,16 +455,17 @@ screen is called the [visual viewport][visual-viewport].
...
 =========
 
-Next up is helping users who prefer darker screens. The reasons why might
-include extra sensitivity to light, or using a device at night, or at night
-near others without disturbing them. For these use cases, browsers these days
-support a *dark mode* feature that darkens the browser and web pages. For
-example, dark mode pages have a black background and a white foreground
-(as opposed to the default white background and black foreground).
+Another useful visual change is using darker colors to help users who
+are extra sensitive to light, or using a device at night, or just
```
Comment (chrishtr) [inline book/accessibility.md:459]:
```suggestion
are extra sensitive to light, use a device at night, or just
```

File: book/accessibility.md [hunk 3]
```diff
@@ -535,26 +539,19 @@ class Browser:
...
-
-Next up is also informing the `Tab` to switch in or out of dark mode:
+[^more-colors]: Of course, a full-featured browser's chrome has many
+    more buttons than our browser's, and probably uses many more
+    buttons. Most browsers support a theming system that stores all
+    the relevant colors and images, and dark mode switches the browser
+    from one theme to another.
+    
+Now, we want the web page content to change from light mode to dark
+mode as well. To start, let's informing the `Tab` when the user
```
Comment (chrishtr) [inline book/accessibility.md:553]:
```suggestion
mode as well. To start, let's inform the `Tab` when the user
```

File: book/accessibility.md [hunk 4]
```diff
@@ -698,6 +638,133 @@ sheets, and instead rely on extensions.
...
+only in dark mode:
+
+
+``` {.css expected=False}
+@media (prefers-color-scheme:dark) {
+  div { background-color: black; color: white; }
+}
+```
+
+Web developers can use `prefers-color-scheme` queries it in their own
```
Comment (chrishtr) [inline book/accessibility.md:665]:
```suggestion
Web developers can use `prefers-color-scheme` queries in their own
```

File: book/accessibility.md (hunk 1 above)
Comment (chrishtr) [reply book/accessibility.md:641]: You do make a good point though that this change reduces redundancy in the chapter...

File: src/lab14.py [hunk 5]
```diff
@@ -445,16 +445,7 @@ def layout(self, zoom):
...
-        if dark_mode:
-            background_color = "black"
-        else:
-            background_color = "white"
-        display_list.append(
-            DrawRect(skia.Rect.MakeLTRB(
-                self.x, self.y, self.x + self.width,
-                self.y + self.height),
-                background_color, background_color))
+    def paint(self, display_list):
```
Comment (chrishtr) [inline src/lab14.py:448]: You didn't indicate this change in your chapter text edits. How about just editing earlier chapters to remove the white background from DocumentLayout's paint method and leave it in draw?

**Comment by chrishtr** [comment]
Per offline discussion, LGTM to commit once the smaller issues I noted are fixed. You can do the rework to put customizations next to the features they customize in another PR.

File: src/lab14.py (hunk 5 above)
Comment (pavpanchekha) [reply src/lab14.py:448]: This is a great find because it turns out we added it in Chapter 13 without explaining it.

**Pull request merged**